firestore/query.go

```diff
@@ -773,7 +773,8 @@ type QuerySnapshotIterator struct {
 // Next blocks until the query's results change, then returns a QuerySnapshot for
 // the current results.
 //
-// Next never returns iterator.Done unless it is called after Stop.
+// Next is not expected to return iterator.Done unless it is called after Stop.
+// Rarely, networking issues may also cause iterator.Done to be returned.
```

Comment:
Is there a way to detect these network issues?

Reply:
Unfortunately I don't think so (except that you might see other weird behavior from RPCs timing out and such). I think the next step for this issue would be to try to disambiguate based on the EOF error, but I'm not sure when we'll have time to get around to that.